src/main.cc

```diff
@@ -470,13 +470,13 @@ int main(int argc, char** argv, char** envp)
 
         log_debug("Datadir is: {}", dataDir.value_or("unset"));
 
-        std::shared_ptr<ConfigManager> configManager;
+        auto configManager = std::make_shared<ConfigManager>(
+            configFile.value_or(""), home.value_or(""), confdir.value_or(""),
```

Comment:
ConfigManager constructor throws `std::runtime_error` in case the file is not readable. This is not catched by your change.

Reply:
Removed.